crates/ra_hir/src/ids.rs

```diff
@@ -171,9 +172,67 @@ impl DefId {
                 Def::Struct(struct_def)
             }
             DefKind::Enum => {
-                let enum_def = Enum::new(self);
+                // FIXME: this isn't great. all this work just to get the `DefId`
+                // of the ancestor `Enum`
+                // TODO: deduplicate!
+                let file_id = loc.source_item_id.file_id;
+                let file_items = db.file_items(file_id);
+                let enum_item_id = loc
+                    .source_item_id
+                    .item_id
+                    .expect("enum node should have item id");
+                let variants = file_items[enum_item_id]
+                    .children()
+                    .find(|node| node.kind() == SyntaxKind::ENUM_VARIANT_LIST)
+                    .unwrap()
```

Comment:
basically any part of the AST can be missing when there are parse errors -- we shouldn't rely on any part being there (in this case for example, we should just construct an `Enum` with no variants)